subsys/bluetooth/audio/has_client.c

```diff
@@ -82,11 +96,166 @@ static enum bt_has_capabilities get_capabilities(const struct has_inst *inst)
 	return caps;
 }
 
+union preset_record {
+	struct bt_has_preset_record_val rec;
+	struct {
+		enum bt_has_properties properties;
+		char name[BT_HAS_PRESET_NAME_MAX + 1]; /* + 1 byte for null-terminator */
+	};
+};
+
+static void handle_read_preset_rsp(struct bt_conn *conn, struct net_buf_simple *buf)
+{
+	struct has_inst *inst = inst_by_conn(conn);
+	const struct bt_has_cp_read_preset_rsp *rsp;
+	union preset_record record;
+	size_t name_len;
+
+	BT_DBG("conn %p buf %p", (void *)conn, buf);
+
+	if (buf->len < sizeof(*rsp)) {
+		BT_ERR("malformed PDU");
+		return;
+	}
+
+	rsp = net_buf_simple_pull_mem(buf, sizeof(*rsp));
```

Comment:
I guess `properties` shall not be verified, because The specification may change so that there will be more property flags defined.
Regarding `is_last` I would not strict to the 0 or 1 even if specification defines those values only. I guess it's sane to provide the preset to the application anyway even if the `is_last` parameter is invalid.

But I guess it would be nice to have at least warning log here.

Reply:
If the core specifies that `is_last` is either 0 or 1, and the server replies with a value other than that, then it's a non-conformant server, and it's pretty risky to consider the remainder of the response as valid IMO. 

Is there a BI test case for it?

Reply:
No, there is no such test case

Reply:
Alright. 

Personally I'd still ignore any indication/notification that contains invalid data (e.g. is_last different from 0 or 1), but the HAP spec does not specify that it shall be ignored directly, and no test case seemingly exist for it (yet), so won't block this PR based on that :)

Reply:
The same question goes for the name length: If the buf->len is e.g. 60, doesn't the imply an invalid preset?